PSR-15 support (No BC Break)

michalbundyra opened the pull request:
This is no BC Break (as long as I can see) because requirements of http-interop/http-middleware in require-dev section was not used and.
I've updated `zend-expressive-helpers` and `zend-expressive-router` dependencies to my PR with PSR-15 support to check if all tests pass here.

I think it will be able to add new new released version of these components and all should be fine.

Requires:
- [x] https://github.com/zendframework/zend-expressive-helpers/pull/50
- [x] https://github.com/zendframework/zend-expressive-router/pull/39

michalbundyra committed:
PSR-15 support

The http-interop/http-middleware dependency in composer require-dev
was not used

michalbundyra committed:
Updated helpers and router dependencies

This is proof that it works with updated components with support PSR-15,
so this package update is not going to have BC Break on migration to PSR-15

weierophinney commented:
We'll need to do a BC break in order to pick up the changes in zendframework/zend-expressive-template#17. I'll create a new release branch in this repo, rebase your PR to it, and push the required changes shortly.

weierophinney committed:
Merge pull request #27 from webimpress/feature/psr-15

PSR-15 support (No BC Break)

Conflicts:
	composer.lock

weierophinney committed:
Updates PlateRenderer to zend-expressive-template 2.0 signatures

Updates all methods to use return types, scalar types, and void types.

weierophinney committed:
Updates doc header and sets strict types

Updates the docheader template to:

- Use https for all links
- add a `declare(strict_types=1)` declaration

Updates all PHP class files to pass the above

weierophinney committed:
Use scalar hints, return type hints, void types, and nullable types

Updates all source code to use PHP 7.1 typehinting.

weierophinney committed:
Remove PHP 5.6 and 7.0 support from CI matrix

weierophinney committed:
Remove redundant jobs from CI build matrix

weierophinney committed:
Update to a later version of PHPUnit

weierophinney commented:
Thanks for getting this started, @webimpress!
